pallets/pallet-nft/src/lib.rs

```diff
@@ -0,0 +1,244 @@
+
+			// Ensure nft data won't over flow.
+			ensure!(data.len() as u32 <= T::MaxSize::get(), Error::<T>::DataTooLarge);
+			ensure!(file_name.len() as u32 <= FILENAME_MAXSIZE, Error::<T>::FileNameTooLarge);
+
+			// Set the Nfts storage
+			let nft_id = Self::next_nft_id();
+
+			PendingNft::<T>::insert(
+				nft_id,
+				(NftData { nft_id, data, file_name: file_name.clone() }, id.clone()),
+			);
+
+			Self::deposit_event(Event::<T>::NFTPending { nft_id, file_name });
+			Ok(())
+		}
+
+		/// Burn an Nft.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::burned())]
+		pub fn burned(origin: OriginFor<T>, nft_id: NftId) -> DispatchResult {
+			// Get the account id
+			let id = ensure_signed(origin)?;
+
+			// Valid nft and owner
+			Self::ensure_nft_is_valid(id.clone(), nft_id)?;
+
+			// Remove storage
+			Nfts::<T>::remove(nft_id);
+			Owners::<T>::remove(nft_id);
+
+			Self::deposit_event(Event::<T>::NftBurned { owner: id, nft_id });
+			Ok(())
+		}
+
+		/// Transfer an Nft.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::transfer())]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			to_user: T::AccountId,
+			nft_id: NftId,
+		) -> DispatchResult {
+			// Get the account id
+			let id = ensure_signed(origin)?;
+
+			// Valid nft and owner
+			Self::ensure_nft_is_valid(id.clone(), nft_id)?;
+
+			// Ensure role is not auditor.
+			ensure!(
+				T::RoleManager::role(&to_user) != Some(Role::Auditor),
+				Error::<T>::IncorrectRole
+			);
+
+			// Transfer Nft ownership to new user.
+			Owners::<T>::mutate(nft_id, |user| {
+				*user = Some(to_user.clone());
+			});
+
+			Self::deposit_event(Event::<T>::NftTransferred { from: id, to: to_user, nft_id });
+			Ok(())
+		}
+
+		/// Audit an Nft pass or fail.
+		///
+		/// Required Auditor.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::approve_nft())]
+		pub fn approve_nft(origin: OriginFor<T>, nft_id: NftId, approve: bool) -> DispatchResult {
+			// Get the account id
+			let id = ensure_signed(origin)?;
+
+			// Ensure auditor role.
+			T::RoleManager::ensure_role(&id, Role::Auditor)?;
+
+			if approve {
+				PendingNft::<T>::take(nft_id)
+					.map(|(nft_data, user)| {
+						Owners::<T>::insert(nft_id, user.clone());
+						Nfts::<T>::insert(nft_id, nft_data);
+						Self::deposit_event(Event::<T>::NftMinted { owner: user, nft_id });
+					})
+					.ok_or(Error::<T>::InvalidNftId)?;
+			} else {
+				PendingNft::<T>::remove(nft_id);
+			}
+
+			Self::deposit_event(Event::<T>::NftAudited { nft_id, approve });
+			Ok(())
+		}
+
+		/// Force burn an Nft.
+		///
+		/// Required governance.
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::force_burn())]
+		pub fn force_burn(origin: OriginFor<T>, nft_id: NftId) -> DispatchResult {
+			// ensure governance
+			T::EnsureGovernance::ensure_origin(origin)?;
+
+			// Remove storage
+			Nfts::<T>::remove(nft_id);
+			if let Some(user) = Owners::<T>::take(nft_id) {
```

Comment:
I think we just need to include nft ID in the event. Don't worry about the owner here.
This simplifies the situation when no owner is found (which shouldn't happen anyway) and no event is logged